template/registry.go

```diff
@@ -20,6 +21,13 @@ type RegistryFunc struct {
 
 type Registry map[string]RegistryFunc
 
+type FuncContext int
+
+const (
+	Format FuncContext = iota
+	Match
+)
```

Comment:
Where are these names coming from? In MF2 spec, I see `Selection` and `Formatting`.

Reply:
Ok, these names are used in our code base.

Reply:
It's coming from the `RegistryFunc` definition
```golang
type RegistryFunc struct {
	Match  func(input any, options Options, locale language.Tag) (output any, err error)
	Format func(input any, options Options, locale language.Tag) (output any, err error)
}
```
but I guess MF2 spec naming would work, too.